### ISSUE TITLE
Find a better way to expose routes to the frontend

### ISSUE THREAD

**Issue by chrabyrd** [archesproject/arches#11346]
The current pattern of `require('arches')` or `import arches from 'arches'` only works because of webpack trickery. It breaks application frontend testing. We should find a better way to expose urls ( and a subset of `settings.py` ) to the frontend.

We could possibly follow the already-in-place pattern of `tsconfig-paths.json`, and create a json file on `app.ready()`

**Comment by jacobtylerwalls** [comment]
> We could possibly follow the already-in-place pattern of tsconfig-paths.json, and create a json file on app.ready()

ready() is fine, but I wonder if we can avoid writing out a file which I feel like was a last resort.

What do you think about exposing an endpoint and then calling it from createVueApplication and awaiting it in parallel with the response for i18n?

**Comment by chrabyrd** [comment]
Ah -- I'd thought about this when first exploring this pattern. I'll admit it's not fantastic to be writing out files in `ready()`, but there's a few reasons why this pattern was chosen. Happy to think/chat about other patterns if they can meet the same reqs.

- `tsconfig-paths.json` is read by `tsconfig.json`. Any time a typescript check is run it must exist and be up-to-date. Eg: pre-commit hook, `npm run ts: check`, CI, and `build_development/build_production`. There's no perfect pattern here, but the current one gets us reasonably close. There's no fantastic way to ping an endpoint for that json as part of the TS process that I'm aware of, but there may be a roundabout way.

- `frontend-configuration-settings.json` is generated with the ( somewhat lofty ) idea that eventually it would be great to run webpack/vite without needing to have the Django server running, cementing the whole separate concerns idea.

- exposing urls to the frontend is tricky, and can't be done via endpoint as far as I'm aware. Django's got various magic incantations it does on urls ( eg `FORCE_SCRIPT_NAME` ), and there's not fantastic way to reliably get the url that would expose the _other_ urls to the frontend.

**Comment by jacobtylerwalls** [comment]
Ah, got it.

re: getting URLs, this might be handy if you haven't seen it: https://forum.djangoproject.com/t/should-the-show-urls-command-be-part-of-djangos-standard-library/20739

**Comment by jacobtylerwalls** [comment]
Duplicate of #11567
